src/backend/executor/nodeShareInputScan.c

```diff
@@ -666,6 +717,29 @@ static int retry_write(int fd, char *buf, int wsize)
 	return 0;
 }
 
+
+
+/*
+ * generate_lock_file_name
+ *
+ * Called by reader or writer to make the unique lock file name.
+ */
+void generate_lock_file_name(char* p, int size, int share_id, const char* name)
+{
+	if (strncmp(name , "writer", 6) == 0)
```

Comment:
My habit is:

strncmp(name , "writer", strlen("writer")) == 0

So you do not need to calculate the string length yourself (compiler does this for you).

Anyway, up to you.